go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -490,7 +490,14 @@ func (tsv *TabletServer) begin(ctx context.Context, target *querypb.Target, save
 		target, options, false, /* allowOnShutdown */
 		func(ctx context.Context, logStats *tabletenv.LogStats) error {
 			startTime := time.Now()
-			if tsv.txThrottler.Throttle() {
+			criticality := tsv.config.TxThrottlerDefaultCriticality
+			if options != nil && options.Criticality != "" {
+				optionsCriticality, err := strconv.Atoi(options.Criticality)
+				if err == nil && optionsCriticality >= 0 && optionsCriticality <= 100 {
+					criticality = optionsCriticality
+				}
```

Comment:
these checks should happen on vtgate. This is too late in the execution path to validate.

Reply:
Added in vtgate and removed from here. I did keep the check for error to ensure we can convert the criticality to an integer.